staging/src/k8s.io/client-go/tools/cache/expiration_cache.go

```diff
@@ -63,26 +63,26 @@ type TTLPolicy struct {
 
 // IsExpired returns true if the given object is older than the ttl, or it can't
 // determine its age.
-func (p *TTLPolicy) IsExpired(obj *timestampedEntry) bool {
-	return p.Ttl > 0 && p.Clock.Since(obj.timestamp) > p.Ttl
+func (p *TTLPolicy) IsExpired(obj *TimestampedEntry) bool {
+	return p.Ttl > 0 && p.Clock.Since(obj.Timestamp) > p.Ttl
 }
 
-// timestampedEntry is the only type allowed in a ExpirationCache.
-type timestampedEntry struct {
-	obj       interface{}
-	timestamp time.Time
+// TimestampedEntry is the only type allowed in a ExpirationCache.
+type TimestampedEntry struct {
```

Comment:
It may be a hard sell to export this method only to use it for caching in the credential provider. We can leave it for now but we may be asked to do something else here.

Reply:
Yea, might have to just store it in a map that is initialized at the start if so like getter or the cachingDockerConfig. Or build my own cache.

Reply:
Wait, by method do you mean struct?

Reply:
If so, I wonder if they'd be more willing to export 
```go
func (c *ExpirationCache) getTimestampedEntry(key string) (*TimestampedEntry, bool)
```

Reply:
cc @cheftako to verify if exporting this struct is going to be an issue

Reply:
Thanks. Basically if it can't be, then no one else can make an IsExpired method.

Reply:
hmmm 🤔 yeah that is a bit odd, it's not even used in other parts of the cache package either

Reply:
fwiw, I don't see `ExpirationCache` being used in all too many places either

Reply:
Might be worthwhile to open a PR now with only the client-go changes to get early feedback from SIG API Machinery.

Reply:
So I think this is fixing an issue in our API, it seems like this should always have been public. That having been said it would be nice to have a warning that it is not safe to share timestamps between computers. You cannot get a timestamp from the API Server and use it on the client machine as part of your expiration cache and expect consistent behavior.